homeassistant/components/homekit/__init__.py

```diff
@@ -149,8 +152,11 @@ def get_accessory(hass, state, aid, config):
         elif device_class == DEVICE_CLASS_ILLUMINANCE or unit in ('lm', 'lx'):
             a_type = 'LightSensor'
 
-    elif state.domain in ('automation', 'input_boolean', 'remote', 'script',
-                          'switch'):
+    elif state.domain == 'switch':
+        switch_type = config.get(CONF_TYPE)
```

Comment:
I agree with changing to `config.get(CONF_TYPE, TYPE_SWITCH)` and removing `None: 'Switch'` from `SWITCH_TYPES`.

I'd prefer to keep the current dict method rather than an if statement. If/when more types are added, it requires fewer lines.